doc/migration.md

```diff
@@ -0,0 +1,358 @@
+| `sf::SoundStream::setLoop`        | `sf::SoundStream::setLooping`     |
+
+## Removal of Default Constructors
+
+The default constructors `sf::Sound::Sound()`, `sf::Text::Text()`, and `sf::Sprite::Sprite()` were removed.
+They can be replaced by the corresponding constructors which accept a resource type.
+
+| v2                     | v3                                         |
+| ---------------------- | ------------------------------------------ |
+| `sf::Sound::Sound()`   | `sf::Sound::Sound(const sf::SoundBuffer&)` |
+| `sf::Text::Text()`     | `sf::Text::Text(const sf::Font&)`          |
+| `sf::Sprite::Sprite()` | `sf::Sprite::Sprite(const sf::Texture&)`   |
+
+Now that these classes are guaranteed to be holding a reference to their corresponding resource type, the functions used to access to those resources now return a reference instead of a pointer.
+These functions are `sf::Sound::getBuffer()`, `sf::Text::getFont()`, and `sf::Sprite::getTexture()`.
+
+## `std::optional` Usage
+
+SFML 3 makes liberal use of C++17's `std::optional` to express when a given function may not return a value.
+Some of these usages have already been mentioned like `sf::WindowBase::pollEvent`.
+Here are some more places where SFML 3 makes use of `std::optional`.
+
+* `sf::IpAddress` uses `std::optional` to express how resolving an address from a string may not yield a usable IP address.
+* `sf::Image::saveToMemory` returns a `std::optional` because the `sf::Image` may be empty or the underlying implementation may fail.
+* `sf::SoundFileReader::open` returns a `std::optional` because the stream being opened may not be valid.
+* `sf::Music::onLoop` and `sf::SoundStream::onLoop` returns a `std::optional` because if the objects are not in a looping state then there is nothing to return.
+* `sf::InputStream` uses `std::optional` in various places. Instead of returning `-1` to signal an error, `std::nullopt` can be returned.
+
+## Fixed Width Integers
+
+SFML 2 contained various typedefs for fixed with integers.
+Those are now replaced with the fixed width integers provided in the `<cstdint>` header from the standard library.
+
+| v2           | v3              |
+| ------------ | --------------- |
+| `sf::Int8`   | `std::int8_t`   |
+| `sf::Uint8`  | `std::uint8_t`  |
+| `sf::Int16`  | `std::int16_t`  |
+| `sf::Uint16` | `std::uint16_t` |
+| `sf::Int32`  | `std::int32_t`  |
+| `sf::Uint32` | `std::uint32_t` |
+| `sf::Int64`  | `std::int64_t`  |
+| `sf::Uint64` | `std::uint64_t` |
+
+## New Constructors for Loading Resources
+
+The following classes gained constructors that allow for loading/opening resources in a single expression.
+Upon failure they throw `sf::Exception`.
+
+* `sf::InputSoundFile`
+* `sf::OutputSoundFile`
+* `sf::Music`
+* `sf::SoundBuffer`
+* `sf::Font`
+* `sf::Image`
+* `sf::RenderTexture`
+* `sf::Shader`
+* `sf::Texture`
+* `sf::FileInputStream`
+* `sf::Cursor`
+
+Here is a before-and-after for loading a sound buffer.
```

Comment:
Even though it's clarified below, it still sounds like a replacement instead of an addition